agora/Agora/Utils.hs

```diff
@@ -115,3 +119,22 @@ puntag ::
   Term s (PTagged tag a) ->
   Term s a
 puntag = pto
+
+{- | Hash the given datum using the correct algorithm(blake2b_256).
+    
+     Note: check the discussion here: https://github.com/input-output-hk/cardano-ledger/issues/2941.
+
+     @since 1.0.0
+-}
+phashDatum ::
+  forall (a :: PType) (s :: S).
+  PIsData a =>
+  Term s (a :--> PDatumHash)
+phashDatum =
+  plam $
```

Comment:
Should we hoist it?

Reply:
Totally forgot about that lol. It's been a long time since I wrote serious plutarch code.